### PR TITLE
implement signup/login logic for error handling

### PR DESCRIPTION
**Pull request by emilyliublair**
## Login/Signup Perms

Issue Number(s): #121 

creates logic for user login and sign up permissions

### Checklist

- [x] Requirements have been implemented
- [x] Acceptance criteria is met
- [x] Database schema [docs](https://www.notion.so/gtbitsofgood/Database-Schema-Docs-fd377ded10b54890b65fd99025488cee?pvs=4) have been updated or are not necessary
- [x] Relevant reviewers (EM) have been assigned to this PR

### Critical Changes

n/a

### Related PRs

PR Number(s): #128 

### How To Test

The steps to test the functionality of your ticket. Include edge cases and testing of error handling, if applicable.

### CODE REVIEW

**Comment by emilyliublair** [comment]
didn't touch callback logic so user will always be directed to /dogs upon clicking login/signup regardless of error or not

File: server/db/actions/User.js [hunk 1]
```diff
@@ -34,27 +34,51 @@ export async function getUserById(id) {
   }
 }
 
-/**
-Updates a user
-@param {*} userId ObjectId of log to update
-@param {*} userData Object with log updates
-*/
-export async function updateUser(userId, userData) {
+export async function updateUser(email, password, name) {
```
Comment (afazio1) [inline server/db/actions/User.js]: make sure changing this functions args doesn't affect other routes that use `updateUser()`. If it does, you can add another function for the user signup use case instead of trying to combine them into 1 function

File: src/pages/api/auth/[...nextauth].js [hunk 2]
```diff
@@ -52,14 +52,40 @@ export const authOptions = {
...
+          credentials.name,
+        );
+
+        if (response.status === 200) {
+          return {
+            id: response.message._id,
+            ...response.message,
+          };
+        } else {
+          return response;
```
Comment (afazio1) [inline src/pages/api/auth/[...nextauth].js]: this error doesn't appear on the frontend see https://stackoverflow.com/questions/76750792/nextauth-how-to-return-custom-errors-without-redirection

File: src/pages/api/auth/[...nextauth].js [hunk 3]
```diff
@@ -68,12 +94,14 @@ export const authOptions = {
...
     signIn: "/login",
-    // newUser: "/dogs",
+    // newUser: "/signup",
     // error: "/login",
   },
   events: {
-    createUser: async (message) => {
-      await dbConnect();
+    updateUser: async (message) => {
+      console.log(message);
```
Comment (afazio1) [inline src/pages/api/auth/[...nextauth].js]: remove console.logs

**Review by afazio1 — CHANGES_REQUESTED** [review]
great work!! See comments for a few changes
